libcloudforensics/providers/gcp/internal/gke.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Google Kubernetes Engine functionalities."""
+
+from typing import TYPE_CHECKING, Any, Dict, Optional
+from libcloudforensics.providers.gcp.internal import common
+
+if TYPE_CHECKING:
+  import googleapiclient
+
+class GoogleKubernetesEngine:
+  """Class to call Google Kubernetes Engine (GKE) APIs.
+
+  https://cloud.google.com/kubernetes-engine/docs/reference/rest
+
+  Attributes:
+    gke_api_client (googleapiclient.discovery.Resource): Client to interact
+        with GKE APIs.
+    project_id (str): Google Cloud project ID.
+  """
+  GKE_API_VERSION = 'v1'
+
+  def __init__(self, project_id: Optional[str] = None) -> None:
+    """Initialize the GoogleCloudStorage object.
+
+    Args:
+      project_id (str): Optional. Google Cloud project ID.
+    """
+
+    self.gke_api_client = None
+    self.project_id = project_id
```

Comment:
`project_id` doesn't seem to be used anywhere?

Reply:
it is not used, just added it by convention as for other services, will probably need it in future functions. Will remove it for now and can add it later